algorithms/gr-bfsrh/cpp/breadth_first_search.cpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <list>
+
+using namespace std;
+
+class Graph {
+  int numVertices;
+  list<int>* adjLists;
+  bool* visited;
+
+   public:
+  Graph(int vertices);
+  void addEdge(int src, int dest);
+  void BFS(int startVertex);
+};
+
+// Create a graph with given vertices,
+// and maintain an adjacency list
+Graph::Graph(int vertices) {
+  numVertices = vertices;
+  adjLists = new list<int>[vertices];

```

Comment:
Codacy found an issue: [Class 'Graph' does not have a copy constructor which is recommended since it has dynamic memory/resource allocation(s).](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6318755)